bundle/libraries/local_path.go

```diff
@@ -80,18 +74,28 @@ func IsLibraryLocal(dep string) bool {
 	return IsLocalPath(dep)
 }
 
-func IsLocalRequirementsFile(dep string) (string, bool) {
-	dep, ok := strings.CutPrefix(dep, "-r ")
-	if !ok {
-		return "", false
+var PipFlagsWithLocalPaths = []string{
+	"-r",
+	"-e",
+}
+
+func IsLocalPathInPipFlag(dep string) (string, string, bool) {
+	for _, flag := range PipFlagsWithLocalPaths {
+		dep, ok := strings.CutPrefix(dep, flag+" ")
+		if ok {
+			dep = strings.TrimSpace(dep)
+			return dep, flag, IsLocalPath(dep)
+		}
 	}
 
-	dep = strings.TrimSpace(dep)
-	return dep, IsLocalPath(dep)
+	return "", "", false
 }
 
 func containsPipFlag(input string) bool {
-	re := regexp.MustCompile(`--[a-zA-Z0-9-]+`)
+	// Trailing space means the the flag takes an argument or there's multiple arguments in input
+	// Alternatively it could be a flag with no argument and no space after it
+	// For example: -r myfile.txt or --index-url http://myindexurl.com or -i
+	re := regexp.MustCompile(`--?[a-zA-Z0-9-]+(\s|$)`)
```

Comment:
This made me realize we also need `^` on the regex (or `^\s*`), or we risk also matching `foo-bar`.

While unlikely to be a valid filename, it is never a flag.

Reply:
true, it should be `^\s*` indeed, will add an use case for this